Daily forecast#6

asmolentzov opened the pull request:
### User Story 6:  Weather Display: Daily Forecast
Closes #6 

- Creates a WeatherDay object to handle weather day forecast information
- Adds weather days forecast table to page
- Adds styling
- Change location image back to being a background image. Styling is much more straightforward. 

#### Review Request
@bendelonlee I would like to refactor/reorganize `index.js` so it's not so cluttered. 
- Any suggestions would be welcome! 
- It sounds from the lessons like it would make sense to start separating some functions out into other files, organized by responsibility. I was thinking maybe one for error handling, and one for the various `post`/`set` functions (e.g. `postWeatherDays`)... that would be a start at least. 
- I have been considering creating a User class but haven't yet and might hold off until I start with login/favoriting... that's next. 

#### Screenshot
![image](https://user-images.githubusercontent.com/6686861/54480868-a4dbbd00-47f3-11e9-9a9a-190da1ab0590.png)

asmolentzov committed:
Adds basic daily forecast structure and styling

asmolentzov committed:
Creates WeatherDay and adds information to index page

asmolentzov committed:
Updates weather days display to table

asmolentzov committed:
Updates table styling

asmolentzov committed:
Reverts location image to being a background image

bendelonlee commented:
This looks really great so far, I'm impressed! You're using object oriented design principles in many crafty ways, and it's a pleasure to read. I do wonder if there's a need to wrap the background image in an object. Right now there's only one attribute. Are you planning to do more than this with this class?

I recommend you go ahead with your plan and put the post / set functions into their own class, and the same with the error checking. Is there a difference between post and set? If not, consolidate into one shared name.

A very small thing - semicolons after function declarations are read as empty statements and serve no purpose, as explained here: https://mariusschulz.com/blog/semicolons-after-function-declarations-in-javascript . During the first half of our lesson, this was explained incorrectly. This issue is very minor, and is unlikely to affect performance in a meaningful way.

asmolentzov committed:
Refactors posting weather method to clean it up

asmolentzov committed:
Moves errors and post handlers to own files

asmolentzov committed:
Removes default data

asmolentzov commented:
@bendelonlee Thank you!! I went ahead and refactored the various post/set methods so they are more consistent in syntax and so that they all use `post` rather than `set` - good call on that. There was no substantive difference in the two, I think just whatever I happened to be doing at the time. 
I also refactored the post functions into one file, and the errors into another - it helps clean up the index.js file. 
Regarding semicolons - looks like the *mind blown* emoji isn't an option here, so just picture it here haha! Thank you for that article!
